src/IIS/Manager/Base/BaseSiteManager.cs

```diff
@@ -1,624 +1,680 @@
                 else
                 {
-                    exists = false;
+                    exists = true;
+                    return site;
                 }
+            }
+            else
+            {
+                exists = false;
+            }
 
 
 
-                //Create Pool
-                ApplicationPoolManager
-                    .Using(_Environment, _Log, _Server)
-                    .Create(settings.ApplicationPool);
+            //Create Pool
+            ApplicationPoolManager
+                .Using(_Environment, _Log, _Server)
+                .Create(settings.ApplicationPool);
 
 
 
-                //Site Settings
-                site = _Server.Sites.Add(
-                    settings.Name,
-                    settings.BindingProtocol.ToString().ToLower(),
-                    settings.BindingInformation,
-                    this.GetPhysicalDirectory(settings));
+            //Site Settings
+            site = _Server.Sites.Add(
+                settings.Name,
+                settings.BindingProtocol.ToString().ToLower(),
+                settings.BindingInformation,
+                this.GetPhysicalDirectory(settings));
 
-                if (settings.CertificateHash != null)
-                {
-                    site.Bindings[0].CertificateHash = settings.CertificateHash;
-                }
+            if (settings.CertificateHash != null)
+            {
+                site.Bindings[0].CertificateHash = settings.CertificateHash;
+            }
 
-                if (!String.IsNullOrEmpty(settings.CertificateStoreName))
-                {
-                    site.Bindings[0].CertificateStoreName = settings.CertificateStoreName;
-                }
+            if (!String.IsNullOrEmpty(settings.CertificateStoreName))
+            {
+                site.Bindings[0].CertificateStoreName = settings.CertificateStoreName;
+            }
 
-                site.ServerAutoStart = settings.ServerAutoStart;
-                site.ApplicationDefaults.ApplicationPoolName = settings.ApplicationPool.Name;
+            site.ServerAutoStart = settings.ServerAutoStart;
+            site.ApplicationDefaults.ApplicationPoolName = settings.ApplicationPool.Name;
 
 
 
-                //Security
-                this.SetAuthentication(settings);
-                this.SetAuthorization(settings);
+            //Security
+            this.SetAuthentication(settings);
+            this.SetAuthorization(settings);
 
-                return site;
-            }
+            return site;
+        }
+
+        protected const string webServerName = "webServer";
+        protected const string ftpServerName = "ftpServer";
+        /// <summary>
+        /// Sets the authentication settings for the site
+        /// </summary>
+        /// <param name="settings">The site settings</param>
+        protected void SetAuthentication(SiteSettings settings)
+        {
+            var authenticationSettings = settings.Authentication;
```

Comment:
I was trying to unify the flow and use only one method for setting authentication - this variable was reused here and there. Nothing major change here.